packages/docker-node/src/docker-manager.ts

```diff
@@ -0,0 +1,162 @@
+import Docker, { ImageBuildContext } from 'dockerode';
+import fsExtra from 'fs-extra';
+import { IncomingMessage } from 'http';
+import {
+  DockerBadGatewayError,
+  DockerHubConnectionError,
+  DockerNotInstalledError,
+  DockerNotRunningError,
+  DockerServerError,
+  ExecutableNotFoundError,
+  ImageDoesntExistError,
+} from './errors';
+import { Image } from './types';
+
+const DOCKER_SOCKET_PATH = '/var/run/docker.sock';
+
+export class DockerManager {
+  public static async create(): Promise<DockerManager> {
+    if (!(await DockerManager.isInstalled())) {
+      throw new DockerNotInstalledError();
+    }
+
+    if (!(await fsExtra.pathExists(DOCKER_SOCKET_PATH))) {
+      throw new DockerNotRunningError();
+    }
+
+    return new DockerManager(Docker);
+  }
+
+  public static async isInstalled(): Promise<boolean> {
+    const { exec } = await import('child_process');
+    return new Promise((resolve) => {
+      exec('which docker', (error?) => resolve(!error));
+    });
+  }
+
+  public static imageToRepoTag(image: Image): string {
+    return `${image.repository}:${image.tag}`;
+  }
+
+  readonly _docker: Docker;
+
+  // The constructor is private, see [[DockerManager.create]].
+  private constructor(DockerImpl: typeof Docker) {
```

Comment:
What're your means to transmit `Docker`? Why not create the `_docker` like this?
```typescript
this._docker = new Docker({ socketPath: DOCKER_SOCKET_PATH })
```